src/jobflow_remote/cli/flow.py

```
@@ -462,6 +462,95 @@ def resume(
+
+    with loading_spinner():
+        with_jobs_info = ["run_dir"]
+        flows_info = jc.get_flows_info(
+            job_ids=job_id,
+            db_ids=db_id,
+            flow_ids=flow_id,
+            states=state,
+            start_date=start_date,
+            end_date=end_date,
+            name=name,
+            metadata=metadata,
+            locked=locked,
+            with_jobs_info=with_jobs_info,
+        )
+
+    if not flows_info:
+        exit_with_warning_msg("No flows matching criteria")
+
+    if not force:
+        if verbosity:
+            preamble = Text.from_markup(
+                f"[red]This operation will [bold]delete the files of the following {len(flows_info)} Flow(s)[/bold][/red]"
+            )
+            out_console.print(preamble)
+            table = get_flow_info_table(flows_info, verbosity=verbosity - 1)
+            out_console.print(table)
+            text = Text.from_markup("[red]Proceed anyway?[/red]")
+        else:
+            text = Text.from_markup(
+                f"[red]This operation will [bold]delete the files of {len(flows_info)} Flow(s)[/bold]. Proceed anyway?[/red]"
+            )
+
+        confirmed = Confirm.ask(text, default=False)
+        if not confirmed:
+            raise typer.Exit(0)
+
+    # if potentially interactive do not start the spinner.
+    spinner_cm: contextlib.AbstractContextManager
+    if jc.project.has_interactive_workers:
+        spinner_cm = contextlib.nullcontext()
+        out_console.print("Deleting files...")
+    else:
+        spinner_cm = loading_spinner(processing=False)
+    with spinner_cm as progress:
+        if progress:
+            progress.add_task(description="Deleting files...", total=None)
+        jobs_info = {}
+        for fi in flows_info:
+            for ji in fi.jobs_info:
+                if ji.run_dir:
+                    jobs_info[ji.db_id] = ji
+        deleted = jc.safe_delete_files(list(jobs_info.values()))
```

This can potentially delete files for jobs that are currently RUNNING no (actually even UPLOADED and SUBMITTED ?) ? Or maybe I missed something somewhere ? If it's the case, shoulnd't there be a confirmation in that case ?

---

This is a good point, but I am unsure how to best deal with this case. 
Should it try to stop the jobs being executed? I think this might be impractical
Is a confirmation just fine? But also, since the jobs are first retrieved and then files deleted, if the runner is active there might be inconsistencies between when the list is retrieved and when the files are actually removed. I would not go to the extent of locking a job though.
Maybe I am overthiking this and a confirmation is fine...

---

I think I would do one the following two options:
1. not allow to delete files for jobs that are UPLOADED, SUBMITTED, or RUNNING
2. allow it but with a confirmation (or some kind of force option ?). That makes me think that maybe we can change the "current" force option above with --yes ? It is just that we directly confirm to delete instead, but we do not force anything right ? This kind of argument (--yes or -y) is also used for other non-jfr commands like apt update or others, which would make it more "easy" to understand. What do you think ?

I'd be in favor of doing 1., it's easier and I think it makes sense to prevent the user from deleting files of a running job.

---

What do you mean exactly for not allow? Should it raise an error, print a warning or just silently skip them? Also there is still the point that unless the job is locked, there is no way of guaranteeing that the job will not switch state between the retrieval of the jobs and the actual deletion of the folders. 

For the `--yes` option, I see your point and it may be good, but the same is used in a bunch of other functions and it would maybe be more confusing to have a different one just for this command alone. If we switch I would say we do for all the commands where the meaning of `-- force` is the same as `--yes`. Indeed there are cases where the meaning is slightly different (e.g. to rerun some kind of jobs). And in that case maybe we should first deprecate the `--force`option by allowing both for a period but warn that the option is deprecated and will be switched to `--yes`

---

I would print a warning mentioning for which jobs the files were not cleaned (or just mentioning that some files were not cleaned ?) and why, e.g. "Cleanup incomplete: files for certain jobs could not be removed because those jobs are in a state that prevents cleaning. Job IDs: [list]."

Regarding the guarantee of switching states between retrieval, I think it's ok, as the "cleanable" states should be downstream states. So basically, the "only" thing that can happen is to "not clean" some files that could have been. (while the opposite, cleaning some files that shouldn't have been cleaned would be more problematic)

For the --yes, let's keep that for another PR then and go through the different cli commands, properly deprecate and do this switch.

---

> Regarding the guarantee of switching states between retrieval, I think it's ok, as the "cleanable" states should be downstream states. So basically, the "only" thing that can happen is to "not clean" some files that could have been. (while the opposite, cleaning some files that shouldn't have been cleaned would be more problematic)

Actually this would be possible. It is not impossible that while the list of jobs is being retrieved the job is being uploaded and so between the retrieval of the list and the start of the cleanup the job switches to the `UPLOADED` state, but for this code the state is still `CHECKED_OUT`, so here it is not even a matter of putting a warning, it is entirely unknown to the code.
To solve such a problem either the runner should be stopped or we need to put a lock on each single job in the flow. I don't see any other way.

Note that the same could actually happen when running `jf flow delete --files`. 

None of these will break the workflow execution, but can potentially leave leftover files. The only point is that it is probably highly unlikely (a user has to decide to delete files or an entire flow while it is still running and at the exact time the runner is working on it), but if we want to take a strict approach the logic should probably be revised to be more rigorous. Considering that deleting Jobs/Flows while they are locked will likely also be somewhat of a pain.

---

Ok from our discussion:

- Only allow deletion of jobs in a COMPLETED/FAILED/REMOTE_ERROR state
- Have a --all-states option to delete files for jobs in any state
- If --all-states is used, only allow when the runner is stopped, unless you use --force
- "Switch" the confirmation from --force to --yes here (other cli commands in another PR)
- Provide the list of jobs for which the files were not deleted (less than X=10?) or ask for confirmation about dumping to a given file with the list of files that were not deleted (case in which we have e.g. 10000 jobs that could not be deleted)

Similar checks could probably be implemented for jf flow delete (another PR, if really needed).

A "separate" thing but somewhat related on the jf flow delete:
- Lock the flow when deleting

---

I should have done all these points.